js/ui/grid_core/ui.grid_core.column_chooser.js

```diff
@@ -318,23 +294,67 @@ const columnChooserMembers = {
 
     _prepareSelectModeConfig: function() {
         const that = this;
-        const selectionChangedHandler = function(e) {
-            const visibleColumns = that._columnsController.getVisibleColumns().filter(function(item) { return !item.command; });
-            const isLastColumnUnselected = visibleColumns.length === 1 && !e.itemData.selected;
+        const selectionOptions = this.option('columnChooser.selection') || {};
 
-            if(isLastColumnUnselected) {
-                e.component.selectItem(e.itemElement);
-            } else {
-                setTimeout(function() {
-                    that._columnsController.columnOption(e.itemData.id, 'visible', e.itemData.selected);
+        const flatNodes = (nodes, result = []) => {
+            nodes.forEach(node => {
+                result.push(node);
+
+                if(node.children.length) {
+                    flatNodes(node.children, result);
+                }
+            });
+
+            return result;
+        };
+
+        const updateSelection = (e, nodes) => {
+            nodes.forEach(node => {
+                if(node.itemData.allowHiding === false) {
+                    e.component.selectItem(node.key);
+                }
+            });
+        };
```

Comment:
Minor comment.

Let's make this function shorter:
```suggestion
        const updateSelection = (e, nodes) => {
          nodes.filter(({ itemData: { allowHiding } }) => allowHiding === false)
                    .forEach(({ key }) => { e.component.selectItem(key); });
        };
```